### ISSUE TITLE
Core dump when deleting an executing code reference

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7328]
Migrated from [rt.perl.org#30027](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=30027) (status was 'resolved')

Searchable as RT30027$

**Comment by p5pRT** [comment]
### From perl-5.8.0@ton.iguana.be

#### Created by perl-5.8.0@ton.iguana.be



\#\!/usr/bin/perl \-w
my %hash;

$hash\{foo\} = sub \{
&nbsp;    delete $hash\{foo\};
&nbsp;    my $buf=5;
&nbsp;    die "Boo";
\};

$hash\{foo\}\->\(\);

Gives&#8203;:

Boo at /home/ton/test\.pl line 7\.
Segmentation fault \(core dumped\)

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.8.2:

Configured by ton at Sun Jan  4 19:19:06 CET 2004.

Summary of my perl5 (revision 5.0 version 8 subversion 2) configuration:
  Platform:
    osname=linux, osvers=2.6.0, archname=i686-linux-64int-ld
    uname='linux quasar 2.6.0 #3 thu dec 18 18:22:48 cet 2003 i686 gnulinux '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=define use64bitall=undef uselongdouble=define
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -fomit-frame-pointer',
    cppflags='-fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='3.4.0 20031231 (experimental)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='long double', nvsize=12, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.3.2.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.3.2'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:



@INC for perl v5.8.2:
    /usr/lib/perl5/5.8.2/i686-linux-64int-ld
    /usr/lib/perl5/5.8.2
    /usr/lib/perl5/site_perl/5.8.2/i686-linux-64int-ld
    /usr/lib/perl5/site_perl/5.8.2
    /usr/lib/perl5/site_perl
    .


Environment for perl v5.8.2:
    HOME=/home/ton
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/ton/bin.Linux:/home/ton/bin:/home/ton/bin.SampleSetup:/opt/schily/bin:/usr/local/bin:/usr/local/sbin:/home/oracle/product/9.0.1/bin:/usr/local/ar/bin:/usr/games/bin:/usr/X11R6/bin:/usr/share/bin:/usr/bin:/usr/sbin:/bin:/sbin:.
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @steve-m-hay

perl\-5\.8\.0@&#8203;ton\.iguana\.be \(via RT\) wrote&#8203;:

>\#\!/usr/bin/perl \-w
>my %hash;
>
>$hash\{foo\} = sub \{
>    delete $hash\{foo\};
>    my $buf=5;
>    die "Boo";
>\};
>
>$hash\{foo\}\->\(\);
>
>Gives&#8203;:
>
>Boo at /home/ton/test\.pl line 7\.
>Segmentation fault \(core dumped\)
>
On my Win32 machine\, this crashes perl\.exe in perl\-5\.8\.2 and perl\-5\.8\.3\, 
but runs fine \(just says "Boo at \.\.\."\, and doesn't crash\) with 
perl\-5\.8\.4 and bleadperl\.

\- Steve





---
Radan Computational Ltd\.

The information contained in this message and any files transmitted with it are confidential and intended for the addressee\(s\) only\.  If you have received this message in error or there are any problems\, please notify the sender immediately\.  The unauthorized use\, disclosure\, copying or alteration of this message is strictly forbidden\.  Note that any views or opinions presented in this email are solely those of the author and do not necessarily represent those of Radan Computational Ltd\.  The recipient\(s\) of this message should check it and any attached files for viruses&#8203;: Radan Computational will accept no liability for any damage caused by any virus transmitted by this email\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From perl5-porters@ton.iguana.be

In article \<40BEFDB6\.5090506@&#8203;uk\.radan\.com>\,
&nbsp;    Steve Hay \<steve\.hay@&#8203;uk\.radan\.com> writes&#8203;:
> On my Win32 machine\, this crashes perl\.exe in perl\-5\.8\.2 and perl\-5\.8\.3\, 
> but runs fine \(just says "Boo at \.\.\."\, and doesn't crash\) with 
> perl\-5\.8\.4 and bleadperl\.
> 
> \- Steve

Ok\, I just compiled a 5\.8\.4\, and can confirm that it also runs without
problems on linux\. So likely this is already solved and can be closed\.

**Comment by p5pRT** [comment]
### From @iabyn

On Thu\, Jun 03\, 2004 at 12&#8203;:04&#8203;:25PM \+0000\, Ton Hospel wrote&#8203;:
> In article \<40BEFDB6\.5090506@&#8203;uk\.radan\.com>\,
>     Steve Hay \<steve\.hay@&#8203;uk\.radan\.com> writes&#8203;:
> > On my Win32 machine\, this crashes perl\.exe in perl\-5\.8\.2 and perl\-5\.8\.3\, 
> > but runs fine \(just says "Boo at \.\.\."\, and doesn't crash\) with 
> > perl\-5\.8\.4 and bleadperl\.
> > 
> > \- Steve
> 
> Ok\, I just compiled a 5\.8\.4\, and can confirm that it also runs without
> problems on linux\. So likely this is already solved and can be closed\.

Yes\, I think it was fixed by change \#22182\, which was integrated into
5\.8\.4\.

\-\- 
The Enterprise's efficient long\-range scanners detect a temporal vortex
distortion in good time\, allowing it to be safely avoided via a minor
course correction\.
&nbsp;    \-\- Things That Never Happen in "Star Trek" \#21
